src/Validation/Constraint/AllowedTo.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation\Constraint;
+
+use DateTimeInterface;
+use Lcobucci\JWT\Validation\ConstraintViolationException;
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\Validation\Constraint;
+
+/**
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ * @since 4.0.0
+ */
+final class AllowedTo implements Constraint
+{
+    /**
+     * @var string
+     */
+    private $audience;
+
+    public function __construct(string $audience)
+    {
+        $this->audience = $audience;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function assert(Token $token)
+    {
+        if (!in_array($this->audience, $token->claims()->get('aud', []), true)) {
+            throw new ConstraintViolationException(
```

Comment:
I suggest a named constructor here. Also, adding the actual values to the exception would be 👍